java/arcs/core/data/expression/Evaluator.kt

```diff
@@ -22,10 +22,12 @@ import arcs.core.util.Time
  * reflected as exceptions.
  */
 class ExpressionEvaluator(
-    val currentScope: Expression.Scope = CurrentScope<Any>(),
+    val currentScope: Expression.Scope = CurrentScope<Any?>(),
     val parameterScope: Expression.Scope = ParameterScope(),
-    val scopeCreator: (String) -> Expression.Scope = { name -> MapScope<Any>(name, mutableMapOf()) }
-) : Expression.Visitor<Any> {
+    val scopeCreator: (String) -> Expression.Scope = {
```

Comment:
By moving this inside, you can no longer supply your own to the evaluator. I guess that's ok for now, but in the future we may want to allow overriding this (e.g. to create RawEntity's directly without copying through a map)

Reply:
You can keep your change, because creating RawEntities directly will require a lot more work, so lets use map copying for now.

Reply:
Oops, I don't think I intended to do this, let me take a look.

Reply:
I didn't move it inside - I just broke some lines as it was >100. I now imported Expression.Scope, so the line is <100 again. Nothing interesting changed here :)